synapseutils/copy.py

```diff
@@ -1,28 +1,80 @@
+
+    :param fileHandles:          List of fileHandle Ids or Objects
+
+    :param associateObjectTypes: List of associated object types: FileEntity, TableEntity, WikiAttachment, UserProfileAttachment, MessageAttachment, TeamAttachment, SubmissionAttachment, VerificationSubmission (Must be the same length as fileHandles)
+    
+    :param associateObjectIds:   List of associated object Ids: If copying a file, the objectId is the synapse id, and if copying a wiki attachment, the object id is the wiki subpage id. (Must be the same length as fileHandles)
+    
+    :param contentTypes:         List of content types (Can change a filetype of a filehandle).
+
+    :param fileNames:            List of filenames (Can change a filename of a filehandle).
+    
+    :return:                     List of batch filehandle copy results, can include failureCodes: UNAUTHORIZED and NOT_FOUND
+    """
+    if (len(fileHandles) != len(associateObjectTypes) or len(fileHandles) != len(associateObjectIds) or
+        len(fileHandles) != len(contentTypes) or len(fileHandles) != len(fileNames)):
+        raise ValueError("Length of fileHandles, associateObjectTypes, and associateObjectIds must be the same")
+    fileHandles = [synapseclient.utils.id_of(handle) for handle in fileHandles]
+    copyFileHandleRequest = {"copyRequests":[]}
+    for filehandleId, contentType, fileName, associateObjectType, associateObjectId in zip(fileHandles, contentTypes, fileNames, associateObjectTypes, associateObjectIds):
+        copyFileHandleRequest['copyRequests'].append({"newContentType":contentType,
+                                                      "newFileName":fileName,
+                                                      "originalFile":{"associateObjectType":associateObjectType,
+                                                                      "fileHandleId":filehandleId,
+                                                                      "associateObjectId":associateObjectId}})
+    copiedFileHandles = syn.restPOST('/filehandles/copy',body=json.dumps(copyFileHandleRequest),endpoint=syn.fileHandleEndpoint)
+    return(copiedFileHandles)
+
+def changeFileMetaData(syn, entity, contentType=None, downloadAs=None):
+    """
+    :param entity:        Synapse entity Id or object
+
+    :param contentType:   Specify content type to change the content type of a filehandle
+
+    :param downloadAs:    Specify filename to change the filename of a filehandle
+
+    :return:              Filehandle copy results, can include failureCodes: UNAUTHORIZED and NOT_FOUND
+    """
+    ent = syn.get(entity,downloadFile=False)
+    if contentType is None:
+        contentType = ent.contentType
+    if downloadAs is None:
+        downloadAs = ent.name
+    copiedFileHandles = copyFileHandles(syn, [ent.dataFileHandleId], [ent.concreteType.split(".")[-1]], [ent.id], [contentType], [downloadAs])
+    return(copiedFileHandles)
+
```

Comment:
This does not change the entity.  If it is changeMetadata it should modify the entity and return.